src/common/perf_counter_proxy.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "common/perf_counters.h"
+#include "common/Mutex.h"
+
+class CephContext;
+/* PerfcounterCollectionProxy hold the lock for PerfCounterCollection */
+class PerfCountersCollectionProxy
```

Comment:
regarding to the naming: how about renaming existing `PerfCountersCollection` to `PerfCountersCollectionImpl` and just extract the locking into `PerfCountersCollection` ?

Reply:
sure, will do